feat: add GIF to figure shortcode

CharlieLeee opened the pull request:
## Add GIF support for figure shortcode

### Features
Add `.gif` to figure shortcode

### Example:
[blog example](https://ckli.xyz/post/cmc/#what-is-contrastive-learning)

rodrigoalcarazdelaosa commented:
Not sure why this is needed. In [this blog post](https://rodrigoalcarazdelaosa.me/en/blog/2020/09/05/color-contrast/) I'm adding a `.gif` with the figure shortcode and everything seems to work just fine.

CharlieLeee commented:
@rodrigoalcarazdelaosa I don't know if this only happens on my side :confused:, but I tried the latest version of wowchemy and created a new post by
```
hugo new --kind post post/new
```
and added the gif file to the post, it did not work
```
{{< figure src="colores-C.gif" title="RGB colors available depending on the desired contrast, ranging from $C=1$ ($256\cdot 256\cdot 256 = 16\thinspace 777\thinspace 216$ available colors) to the limit $C_\text{b\&w}^\text{max}=\sqrt{21}$ in which we only have the fuchsia color #cf0dcc." lightbox="false" width="100%" >}}
```
![image](https://user-images.githubusercontent.com/33972674/92557914-f9a0ac80-f29f-11ea-8057-144b06599ce9.png)
and once I modified the `figure.html` this way, it worked out fine
![image](https://user-images.githubusercontent.com/33972674/92558201-8ea3a580-f2a0-11ea-811f-7b169d17d7d9.png)
